config/http.go

```diff
@@ -3,63 +3,55 @@ package config
 import (
 	"context"
 	"crypto/tls"
+	"net"
 	"net/http"
-	"sync/atomic"
+	"sync"
 
 	"github.com/pomerium/pomerium/internal/log"
 	"github.com/pomerium/pomerium/internal/tripper"
 	"github.com/pomerium/pomerium/pkg/cryptutil"
-
-	"github.com/rs/zerolog"
 )
 
-type httpTransport struct {
-	underlying *http.Transport
-	transport  atomic.Value
-}
-
 // NewHTTPTransport creates a new http transport. If CA or CAFile is set, the transport will
 // add the CA to system cert pool.
-func NewHTTPTransport(src Source) http.RoundTripper {
-	ctx := log.WithContext(context.TODO(), func(c zerolog.Context) zerolog.Context {
-		return c.Caller()
-	})
-	t := new(httpTransport)
-	t.underlying, _ = http.DefaultTransport.(*http.Transport)
-	src.OnConfigChange(ctx, func(ctx context.Context, cfg *Config) {
-		t.update(ctx, cfg.Options)
-	})
-	t.update(ctx, src.GetConfig().Options)
-	return t
-}
-
-func (t *httpTransport) update(ctx context.Context, options *Options) {
-	nt := new(http.Transport)
-	if t.underlying != nil {
-		nt = t.underlying.Clone()
-	}
-	if options.CA != "" || options.CAFile != "" {
-		rootCAs, err := cryptutil.GetCertPool(options.CA, options.CAFile)
-		if err == nil {
-			nt.TLSClientConfig = &tls.Config{
-				RootCAs:    rootCAs,
-				MinVersion: tls.VersionTLS12,
+func NewHTTPTransport(src Source) *http.Transport {
+	var (
+		lock      sync.Mutex
+		tlsConfig *tls.Config
+	)
+	update := func(ctx context.Context, cfg *Config) {
+		if cfg.Options.CA != "" || cfg.Options.CAFile != "" {
+			rootCAs, err := cryptutil.GetCertPool(cfg.Options.CA, cfg.Options.CAFile)
+			if err == nil {
+				lock.Lock()
+				tlsConfig = &tls.Config{
+					RootCAs:    rootCAs,
+					MinVersion: tls.VersionTLS12,
+				}
+				lock.Unlock()
+			} else {
+				log.Error(context.Background()).Err(err).Msg("config: error getting cert pool")
```

Comment:
```suggestion
				log.Error(ctx).Err(err).Msg("config: error getting cert pool")
```

as it'll carry information which config change caused it.